include/emp/prefab/ControlPanel.hpp

```diff
@@ -0,0 +1,331 @@
+          &do_redraw=Info()->GetRedrawChecker()]
+        (const web::Animate & anim) mutable {
+          // Run the simulation function every frame
+          run_sim();
+          // Redraw widgets according to a rule
+          if(do_redraw(anim)) {
+            for (auto & wid : refresh_list) {
+              wid.Redraw();
+            }
+          }
+        }
+      );
+
+      toggle_run.SetCallback(
+        [&anim=Animate(GetID()), step=web::Button(step)]
+        (bool is_active) mutable {
+          if (is_active) {
+            anim.Start();
+          } else {
+            anim.Stop();
+          }
+        }
+      );
+
+      step.SetCallback([&anim=Animate(GetID())]() {
+        anim.Step();
+      });
+    }
+
+    public:
+    /**
+     * Contructor for a Control panel.
+     * @param refresh_rate the number of milliseconds or frames between refreshes
+     * @param refresh_mode units of "MILLISECONDS" or "FRAMES"
+     * @param in_id HTML ID of control panel div
+     */
+    ControlPanel(
+      const int & refresh_rate,
+      const std::string & refresh_unit,
+      const std::string & in_id="")
+    : ControlPanel(
+      refresh_rate,
+      refresh_unit,
+      new internal::ControlPanelInfo(in_id)
+    ) { ; }
+
+    /**
+     * Set the simulation for this control panel
+     * @param sim the function to be run every frame (as fast as possible)
+     */
+    ControlPanel & SetSimulation(const std::function<void()> & sim) {
+      Info()->SetSimulation(sim);
+      return *this;
+    }
+
+    /**
+     * Get the simulation for this control panel
+     * @return the function to be run every frame (as fast as possible)
+     */
+    const std::function<void()> & GetSimulation() const {
+      return Info()->GetSimulation();
+    }
+
+    /**
+     * Set the refresh rate units for this control panel.
+     * @param unit either "MILLISECONDS" or "FRAMES"
+     * @note rates are independent for "MILLISECONDS" and "FRAMES"
+     */
+    ControlPanel & SetRefreshUnit(const std::string & units) {
+      Info()->SetUnit(units);
+      return *this;
+    }
+
+    /**
+     * Set the refresh rate for this control panel for the current unit.
+     * @param rate period in frames or milliseconds
+     * @note rates are independent for "MILLISECONDS" and "FRAMES".
+     */
+    void SetRefreshRate(const int & rate) {
+      Info()->SetRate(rate);
+    }
+
+    /**
+     * Set the refresh rate for this control panel.
+     * @param rate the number of milliseconds or frames between refreshes
+     * @param unit either "MILLISECONDS" or "FRAMES"
+     * @note rates are independent for "MILLISECONDS" and "FRAMES"
+     */
+    void SetRefreshRate( const int & rate, const std::string & units) {
+      Info()->SetUnit(units);
+      Info()->SetRate(rate);
+    }
+
+    /**
+     * Adds a Widget to a list of widgets redrawn at the specified refresh rate.
+     * @param area a widget
+     */
+    void AddToRefreshList(Widget & area) {
```

Comment:
Is there a definite use case for public access to this? We can make it private and then always make it public later